qiita_db/meta_util.py

```diff
@@ -555,20 +555,44 @@ def generate_plugin_releases():
         f(redis_key, v)
 
 
-def get_software_commands(active):
+def _build_software_commands_col_names_object(active):
+    '''
+    Helper function for update_resource_allocation_redis
+
+    Constructs a nested dictionary containing software commands and associated
+    column names from the database.
+    '''
+
+    col_name_list = []
+    with qdb.sql_connection.TRN:
+        sql = ''' SELECT * FROM qiita.resource_allocation_column_names; '''
+        qdb.sql_connection.TRN.add(sql)
+        res = qdb.sql_connection.TRN.execute_fetchindex()
+
+    for col_name in res:
+        col_name_list.append(col_name[1])
+
```

Comment:
The sql above selects all columns but you only use [1], could you make sure to just select the column you want and use fetch_flatten to get only the values you need?